File: frontend/src/app/services/group.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {Group} from "../models/Group";
import {User} from "../models/User";

@Injectable({
  providedIn: 'root',
})
export class GroupService {
  private apiUrl = 'http://localhost:8080/api/groups';

  constructor(private http: HttpClient) {}

  getAllGroups(): Observable<Group[]> {
    return this.http.get<Group[]>(this.apiUrl);
  }

  createGroup(group: Group): Observable<Group> {
    return this.http.post<Group>(this.apiUrl, group);
  }

  updateGroup(groupId: number, group: Group): Observable<Group> {
    return this.http.put<Group>(`${this.apiUrl}/${groupId}`, group);
  }

  deleteGroup(groupId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${groupId}`);
  }

  getGroupMembers(groupId: number): Observable<User[]> {
    return this.http.get<User[]>(`${this.apiUrl}/${groupId}/members`);
  }

  addMembersToGroup(groupId: number, userIds: number[]): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/${groupId}/members`, userIds);
  }
}

File: backend/src/main/java/edu/miage/springboot/services/interfaces/GroupService.java
package edu.miage.springboot.services.interfaces;

import edu.miage.springboot.dao.entities.GroupEntity;
import edu.miage.springboot.dao.entities.UserEntity;

import java.util.List;
import java.util.Set;

public interface GroupService {
    List<GroupEntity> getAllGroups();
    GroupEntity getGroupById(Long id);
    GroupEntity createGroup(GroupEntity group);
    GroupEntity updateGroup(Long id, GroupEntity groupDetails);
    boolean deleteGroup(Long id);
    void addMembersToGroup(Long groupId, List<Long> userIds);
    Set<UserEntity> getGroupMembers(Long groupId);
}
